### PR TITLE
Binance create_order with REST api

### PR DESCRIPTION
**Pull request by m-an**

### CODE REVIEW

File: lib/arke/exchange/binance.rb [hunk 1]
```diff
@@ -16,6 +17,10 @@ def initialize(opts)
 
       @orderbook = Arke::Orderbook.new(@market)
       @last_update_id = 0
+      @rest_api_connection = Faraday.new("https://api.binance.com/api/v3") do |builder|
```
Comment (mod) [inline lib/arke/exchange/binance.rb]: Use host from config

File: lib/arke/exchange/base.rb [hunk 2]
```diff
@@ -17,5 +17,23 @@ def initialize(opts)
       rate_limit = 1.0 if rate_limit <= 0
       @min_delay = 1.0 / rate_limit
     end
+
+    def build_error(response)
```
Comment (mod) [inline lib/arke/exchange/base.rb:21]: this helper don't make much sense, also parsing json twice

File: lib/arke/exchange/binance.rb [hunk 3]
```diff
@@ -65,5 +70,35 @@ def get_snapshot
       process(snapshot['bids'], :buy)
       process(snapshot['asks'], :sell)
     end
+
+    def create_order(order)
+      request = @rest_api_connection.post("order") do |req|
```
Comment (mod) [inline lib/arke/exchange/binance.rb]: create a post method with apikeys

File: spec/arke/exchange/binance_spec.rb [hunk 4]
```diff
@@ -105,4 +116,30 @@
...
+    let(:order) { Arke::Order.new('ETHUSDT', '250', '1', :buy) }
+    let(:timestamp) { "1551720218" }
+    let(:query) do
+      "symbol=#{order.market}&side=#{order.side.upcase}&type=LIMIT&timeInForce=GTC&quantity=#{order.amount.to_f}"\
+        "&price=#{order.price.to_f}&recvWindow=5000&timestamp=#{timestamp}"
+    end
+    let(:signature) do
+      OpenSSL::HMAC.hexdigest(OpenSSL::Digest.new('sha256'), 'OwpadzSYOSkzweoJkjPrFeVgjOwOuxVHk8FXIlffdWw', query)
+    end
+
```
Comment (mod) [inline spec/arke/exchange/binance_spec.rb:143]: are you using webmock ? where do you mock the binance api ?

**Review by mod — CHANGES_REQUESTED** [review]
almost done

File: lib/arke/exchange/base.rb (hunk 2 above)
Comment (m-an) [reply lib/arke/exchange/base.rb:21]: I've just copypasted it from `Rubykube` exchange, I can redo

File: spec/arke/exchange/binance_spec.rb (hunk 4 above)
Comment (m-an) [reply spec/arke/exchange/binance_spec.rb:143]: in next file, `mocked_context.rb`

**Review by mod — APPROVED** [review]

**Pull request merged**